inspire_disambiguation/api.py

```diff
@@ -221,30 +210,137 @@ def cluster(
             ),
             signature_block=phonetic_block,
             B3_f_score=cluster.supervised_scoring(clusterer.y, cluster.labels_)
-            if hasattr(cluster, "supervised_scoring") and not test_labels
+            if hasattr(cluster, "supervised_scoring")
             else None,
         )
-    if test_labels:
-        statistics_names = ("precision", "recall", "f1")
-        training_statistics = clusterer.score(test_signatures_uuids, test_labels)
-        (
-            (B3_statistics_all, wrongly_classified_pairs),
-            B3_statistics_training,
-            B3_statistics_test,
-        ) = training_statistics
+    return process_clustering_output(clusterer)
+
+
+def cluster_with_evaluation(
+    ethnicity_model_path, distance_model_path, n_jobs, test_signatures_uuids=None,
+):
+    """Train the clustering model and process the output.
+
+    Args:
+        ethnicity_model_path (str): Full path where ethnicity model is saved.
+        distance_model_path (str): Full path where distance model is saved.
+        n_jobs (int): Number of processes to use.
+        test_signatures_uuids (set): Signature uuids which will be used
+        for model validation.
+    """
+    start_time = datetime.now()
+    LOGGER.info("Preparing test dataset...")
+    signature_blocks = get_signature_blocks(only_curated=True)
+    labels_train, labels_test, y_train, y_test = (
+        np.array([]),
+        np.array([]),
+        np.array([]),
+        np.array([]),
+    )
+    clustered_blocks = 0
+    statistics_names = ("precision", "recall", "f1")
+    for block in signature_blocks:
+        clustered_blocks += 1
         LOGGER.info(
-            B3_precision_recall_f_score_all=dict(
-                zip(statistics_names, B3_statistics_all)
-            ),
-            B3_precision_recall_f_score_training=dict(
-                zip(statistics_names, B3_statistics_training)
-            ),
-            B3_precision_recall_f_score_test=dict(
-                zip(statistics_names, B3_statistics_test)
-            ),
-            wrongly_classified_pairs=wrongly_classified_pairs,
+            f"""Clustering {clustered_blocks} out of {len(signature_blocks) + 1}"""
```

Comment:
We're doing structured logging, which means you shouldn't interpolate strings here but use a constant string for information and pass the variable data as keyword args. Something like
```suggestion
            "Clustering a new block",  current=clustered_blocks, total=(len(signature_blocks) + 1)
```